src/run.js

```diff
@@ -1,62 +1,33 @@
-const { dirname, join, posix, relative, sep } = require('path');
-const { readFileSync } = require('graceful-fs');
-const { parse } = require('jest-docblock');
-const tsd = require('mlh-tsd');
-const formatErrorMessage = require('./formatErrorMessage');
-const { pass } = require('./pass');
+const tsdLite = require('tsd-lite');
+const { formatTsdErrors, formatTsdResults } = require('./formatter');
 const { fail } = require('./fail');
+const { pass } = require('./pass');
 
 const TEST_TITLE = 'tsd typecheck';
 
-/**
- * @param {string} testFile
- * @param {string} fileContents
- */
-function resolveTypingsFile(testFile, fileContents) {
-  let { type } = parse(fileContents);
-
-  if (Array.isArray(type)) {
-    type = type[0];
-  }
-
-  if (type !== undefined) {
-    return join(dirname(testFile), type);
-  }
-
-  return undefined;
-}
-
-/**
- * @param {string} input
- */
-const normalizeSlashes = input => input.split(sep).join(posix.sep);
-
-module.exports = async ({ config: { rootDir }, testPath }) => {
-  const testFileContents = readFileSync(testPath, 'utf8');
-
-  const testFile = relative(rootDir, testPath);
-  const typingsFile = resolveTypingsFile(testFile, testFileContents);
-
+module.exports = async ({ testPath }) => {
   const start = Date.now();
 
-  const { diagnostics, numTests } = await tsd.default({
-    cwd: rootDir,
-    testFiles: [normalizeSlashes(testFile)],
-    typingsFile,
-  });
+  const { assertionsCount, tsdErrors, tsdResults } = tsdLite.default(testPath);
 
   const end = Date.now();
 
-  const numFailed = diagnostics.length;
-  const numPassed = numTests - numFailed;
+  if (tsdErrors !== undefined) {
+    const message = formatTsdErrors(tsdErrors);
+
+    throw { message, stack: '' };
```

Comment:
Strange line. I was looking at Jest code, this was enough to print nice message with `Test suite failed to run` title on top. Alternatively it is possible to throw an error in `tsd-lite`. The result would be almost the same: if errors are formatted here, it is possible to report several errors at once (see tests); if `tsd-lite` would throw, only one error could be reported.

Reply:
I will rework this. Tried it out, all shapes up much better if `tsd-lite` throws an error.